### ISSUE TITLE
Bug: null used as key in "~#cmap"

### ISSUE THREAD

**Issue by wevre** [cognitect/transit-java#37]
If `null` is used as a key in a "~#cmap", it will not be read correctly.

To reproduce, make the following changes in `transit-format`:

1. Add an exemplar in `transit.exemplar` (around line 115):

```clj
(write-exemplar dir "cmap_null_key" "Cmap with null key" {nil "null as map key" [1 2] "Array as key to force cmap"})
```

2. Regenerate the exemplars by running `bin/exemplars`.

3. Run `bin/verify -impls java`.

The problem arises because the internal [ArrayReader](https://github.com/cognitect/transit-java/blob/master/src/main/java/com/cognitect/transit/impl/ReadHandlers.java#L56) for the CmapReadHandler uses `null` as a flag to toggle back and forth between storing a key and storing an item. The implementation for transit-clj, on the other hand, does not have this bug because it uses a singleton `(Object.)` as a [marker](https://github.com/cognitect/transit-clj/blob/master/src/cognitect/transit.clj#L244).

**Comment by puredanger** [comment]
Thanks for the report!

**Comment by puredanger** [comment]
Fixed in transit-java 1.0.366
